starter-code/routes/auth.js

```diff
@@ -0,0 +1,92 @@
+const express = require("express");
+const bcrypt = require("bcrypt");
+const passport = require("passport");
+const ensureLogin = require("connect-ensure-login");
+
+const User = require("../models/user");
+
+const router = express.Router();
+const bcryptSalt = 10;
+
+// PRIVATE
+router.get("/private-page", ensureLogin.ensureLoggedIn(), (req, res) => {
+  res.render("passport/private", { user: req.user });
+});
+
+// LOGIN
+router.get("/login", (req, res, next) => {
+  res.render("passport/login");
+});
+
+router.post(
+  "/login",
+  passport.authenticate("local", {
+    successRedirect: "/private-page",
+    failureRedirect: "/login",
+    failureFlash: true,
```

Comment:
You are setting "failureFlash" to true but you are not passing any error to the view (see previous comment).